src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1,32 +1,196 @@
 package edu.umich.its.cpm;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.concurrent.atomic.AtomicLong;
 
+import javax.servlet.http.HttpServletRequest;
+
+import java.util.Map;
+import java.net.URI;
+import java.net.URISyntaxException;
+
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.core.env.Environment;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.client.RestClientException;
 
-import com.google.gson.Gson;
+import org.json.JSONObject;
+import org.json.JSONArray;
 
 @RestController
 public class MigrationController {
 
-    private static final String template = "Hello, %s!";
+	private static final Logger log = LoggerFactory.getLogger(MigrationController.class);
+	
     private final AtomicLong counter = new AtomicLong();
     
     @Autowired MigrationRepository repository;
+    
+    @Autowired
+    private Environment env;
+    
+    /**
+     * get all CTools sites where user have site.upd permission 
+     * @return
+     */
+    @RequestMapping("/projects")
+    public String getProjectSites() {
+    	String rv = "";
+    	
+    	// login to CTools and get sessionId
+    	String sessionId = login_becomeuser();
+    	if (sessionId != null)
+    	{
+			// get all sites that user have permission site.upd
+			RestTemplate restTemplate = new RestTemplate();
+			// the url should be in the format of "https://server/direct/site/withPerm/.json?permission=site.upd"
+			String requestUrl = env.getProperty("ctools.direct.url") + "site/withPerm/.json?permission=site.upd&_sessionId=" + sessionId;
```

Comment:
Why is the format 'site/withPerm/.json?...'?  If json is a format isn't it attached to an URL element not a separate element?